p2p/src/peer_manager/mod.rs

```diff
@@ -1515,7 +1522,8 @@ where
                 // that can have `discovered_own_address`.
                 let delay = (RESEND_OWN_ADDRESS_TO_PEER_PERIOD / OUTBOUND_FULL_RELAY_COUNT as u32)
                     .mul_f64(utils::exp_rand::exponential_rand(&mut make_pseudo_rng()));
-                next_time_resend_own_address += delay;
+                next_time_resend_own_address = (next_time_resend_own_address + delay)
+                    .expect("Time derived from local clock; cannot fail");
```

Comment:
Again, `AddAssign` would be nice to have.

Reply:
Doesn't work with `Option<_>`